altapay.php

```diff
@@ -358,66 +360,73 @@ function altapayCaptureCallback() {
 				}
 
 				$orderLines = $utilMethods->createOrderLines( $order, $selectedProducts );
-				$salesTax   = 0;
-				foreach ( $orderLines as $orderLine ) {
-					$salesTax += $orderLine['taxAmount'];
-				}
-			} else {
-				$orderLines = array( array() );
-				$salesTax   = $order->get_total_tax();
 			}
 
-			// Capture amount
-			$captureResult = $api->captureReservation( $txnID, $amount, $orderLines, $salesTax );
-			if ( ! $captureResult->wasSuccessful() ) {
-				// log to history
-				$order->add_order_note( __( 'Capture failed: ' . $captureResult->getMerchantErrorMessage(), 'Altapay' ) );
-				echo wp_json_encode(
-					array(
-						'status'  => 'error',
-						'message' => $captureResult->getMerchantErrorMessage(),
-					)
-				);
-			} else {
-				// Get payment data
-				$payment = $api->getPayment( $txnID );
-				if ( $payment ) {
-					$payments = $payment->getPayments();
-					foreach ( $payments as $pay ) {
-						$reserved = $pay->getReservedAmount();
-						$captured = $pay->getCapturedAmount();
-						$refunded = $pay->getRefundedAmount();
-						$charge   = $reserved - $captured - $refunded;
-						if ( $charge <= 0 ) {
-							$charge = 0.00;
-						}
-					}
+			try {
+				$api = new CaptureReservation( $settings->getAuth() );
+				$api->setAmount( (float) number_format( $amount, 2, '.', '' ) );
+				$api->setOrderLines( $orderLines );
+				$api->setTransaction( $txnID );
+				$response = $api->call();
+			} catch ( InvalidArgumentException $e ) {
+				error_log( 'Response header exception ' . $e->getMessage() );
+			} catch ( ResponseHeaderException $e ) {
+				error_log( 'Response header exception ' . $e->getMessage() );
+			} catch ( \Exception $e ) {
+				error_log( 'Response header exception ' . $e->getMessage() );
+			}
+
+			if ( $response->Result !== 'Success' ) {
+				wp_send_json_error( array( 'error' => __( 'Could not capture reservation' ) ) );
+			}
+
+			$rawResponse = $api->getRawResponse();
+			$charge      = 0;
+			$reserved    = 0;
+			$captured    = 0;
+			$refunded    = 0;
+
+			if ( ! empty( $rawResponse ) ) {
+				$body = $rawResponse->getBody();
+				// Update comments if capture fail
+				$xml = simplexml_load_string( $body );
```

Comment:
create a `SimpleXMLElement` object directly instead as that will give us better error handling
```suggestion
				$xml = new SimpleXMLElement( $body );
```